### DISCUSSION TITLE
Distributed areas & CIM classes

### DISCUSSION THREAD

**Discussion by AAndersn** [GRIDAPPSD/gridappsd-forum#148] [Ideas]
Discussion thread for possible distributed model classes that correspond to CIM classes

Classes currently being used as equipment containers in different CIM models we need to parse:

* `ConnectivityNodeContainer` (bus-branch models)
* `GeographicalRegion` (all)
* `SubGeographicalRegion` (all)
* `Substation` (distribution & node-breaker)
* `VoltageLevel` (node-breaker)
* `Feeder` (distribution)

Extensions under consideration
* `SwitchArea` or `DistributionArea`
* `SecondaryArea`

![image](https://user-images.githubusercontent.com/19935503/229884624-1e9b64d0-0eb4-4c89-b2ba-fe55d944e871.png)

**Comment by craigpnnl** [comment]
Weren't @AAndersn and @afisher1 talking about DistributedModel and you wanted DistributedFeeder and DistributedTransimision?

**Reply by AAndersn** [reply]
Correct. Concept for the CIM-Graph parent python classes would be:

* `NodeBreakerModel` -- contains Substation
* `DistributedTransmissionModel` -- contains Substation, VoltageLevel, Feeder, SwitchArea, SecondaryArea
* `BusBranchModel` -- contains ConnectivityNodeContainer, ?? (TBD, see https://github.com/GRIDAPPSD/CIMHub/blob/feature/SETO/BES/IEEE118_CIM.xml)
* `CentralizedFeederModel` -- contains Feeder
* `DistributedFeederModel` -- contains Feeder, SwitchArea, SecondaryArea

Child classes, may need to avoid confusion with similarly named data classes
* `Substation` -- contains VoltageLevel, Feeder, SwitchArea, SecondaryArea, may need to avoid confusion with cim.Substation data class
* `Feeder` -- contains SwitchArea, Secondary Area
* `SwitchArea` -- contains Secondary Area
* `SecondaryArea` -- contains no lower classes

Current structure:

![image](https://user-images.githubusercontent.com/19935503/229892414-9757c1f9-6459-4408-bd92-b12e8cba470e.png)

Possible new structure (TBD, focus of this thread):

![image](https://user-images.githubusercontent.com/19935503/229893719-c3e20e48-eeb3-4062-a57f-53eeb93b8655.png)
